client/app/models/contact.coffee

```diff
@@ -148,6 +148,26 @@ module.exports = class Contact extends Backbone.Model
         _.compact([pf, fn, mn, gn, sf]).join ' '
 
 
+    # We override toJSON to ensure the exported object is the a true object.
+    # Allow a options.toVCF boolean to sanitize the output before.
+    toJSON: ({toVCF} = {}) ->
+        data = super
+        data.datapoints = data.datapoints.toJSON()
+
+        return data unless toVCF
+
+        # Isolate in a try..catch block to go outside main thread and pop error
+        try
+            # We remove the BDay datapoint if it doesn't match ISO 8601
+            unless moment(data.bday, 'YYYY-MM-DD', true).isValid()
+                delete data['bday']
+                throw new Error "Malformed bday for #{data.id}"
```

Comment:
I think that its useless to throw an error in this try catch block (except if moment() could throw another one ?), what do you think about directly call the `console.error` method, and remove the try/catch block ?